src/app/lib/utilities/preprocessor/rules/cpsAssetPage/addBylineBlock/index.js

```diff
@@ -0,0 +1,49 @@
+import pathOr from 'ramda/src/pathOr';
+import path from 'ramda/src/path';
+import deepClone from 'ramda/src/clone';
+import insertBlockAfterHeadline from '../helpers';
+
+const getBylineBlock = json => {
+  const byline = pathOr(null, ['promo', 'byline'], json);
+
+  if (!byline) {
+    return null;
+  }
+
+  const bylineBlock = {
+    model: {
+      blocks: [
+        {
+          name:
+            path(['name'], byline) || path(['persons', '0', 'name'], byline),
```

Comment:
Should we use `pathOr`'s for these like the code above? And add some default values if these aren't available?

Reply:
thanks @HarveyPeachey, I had concerns with these since it's an optional maybe using pathOr for the second bit like 
```
path(['name'], byline) || pathOr(null, ['persons', '0', 'name'], byline), 
```